src/main/kotlin/telegram/Handling.kt

```diff
@@ -385,35 +362,32 @@ object MafiaHandler {
                 showAdMenu(chat, bot)
             }
             simple(adNewCommand) {
-                val chat = ChatId.fromId(chatId)
-                val res = bot.sendMessage(
-                    chat,
+                val msgId = sendMessage(
+                    bot,
+                    chatId,
                     Const.Message.enterAdText
                 )
-                if (res.isSuccess) {
-                    val msgId = res.get().messageId
-                    bot.editMessageReplyMarkup(
-                        chat,
-                        msgId,
-                        replyMarkup = inlineKeyboard {
-                            button(closePopupCommand, chatId)
-                        }
-                    )
+                if (msgId != -1L) {
                     adPopups.save(AdPopup(ObjectId(), chatId, msgId))
+                    updateMessage(
+                        bot, chatId, msgId,
+                        inlineKeyboard { button(closePopupCommand, chatId) }
+                    )
                 }
             }
             simple(adminCommand, adminPanelCommand, adminPanelLegacyCommand) {
                 if (messageId != null) {
                     bot.deleteMessage(ChatId.fromId(chatId), messageId)
                 }
-                val res = bot.sendMessage(
-                    ChatId.fromId(chatId),
-                    "Меню администратора:"
+                showAdmin(
+                    chatId,
+                    sendMessage(
```

Comment:
Two separate actions in one line, easy to miss that there is a new message being sent